Assets/Scripts/Models/InventoryManager.cs

```diff
@@ -6,100 +6,89 @@
 // file LICENSE, which is part of this source code package, for details.
 // ====================================================
 #endregion
-
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using MoonSharp.Interpreter;
-using UnityEngine;
 
 [MoonSharpUserData]
 public class InventoryManager
 {
-    // This is a list of all "live" inventories.
-    // Later on this will likely be organized by rooms instead
-    // of a single master list. (Or in addition to.)
-    public Dictionary<string, List<Inventory>> inventories;
+    private static readonly string InventoryManagerLogChanel = "InventoryManager";
 
     public InventoryManager()
     {
-        inventories = new Dictionary<string, List<Inventory>>();
+        Inventories = new Dictionary<string, List<Inventory>>();
     }
 
-    public bool PlaceInventory(Tile tile, Inventory inv)
+    public Dictionary<string, List<Inventory>> Inventories { get; private set; }
+
+    public bool PlaceInventory(Tile tile, Inventory inventory)
     {
         bool tileWasEmpty = tile.Inventory == null;
 
-        if (tile.PlaceInventory(inv) == false)
+        if (tile.PlaceInventory(inventory) == false)
         {
             // The tile did not accept the inventory for whatever reason, therefore stop.
             return false;
         }
 
-        CleanupInventory(inv);
+        CleanupInventory(inventory);
 
-        // We may also created a new stack on the tile, if the tile was previously empty.
+        // We may also created a new stack on the tile, if the startTile was previously empty.
         if (tileWasEmpty)
         {
-            if (inventories.ContainsKey(tile.Inventory.ObjectType) == false)
+            if (Inventories.ContainsKey(tile.Inventory.ObjectType) == false)
             {
-                inventories[tile.Inventory.ObjectType] = new List<Inventory>();
+                Inventories[tile.Inventory.ObjectType] = new List<Inventory>();
             }
 
-            inventories[tile.Inventory.ObjectType].Add(tile.Inventory);
+            Inventories[tile.Inventory.ObjectType].Add(tile.Inventory);
 
             World.Current.OnInventoryCreatedCallback(tile.Inventory);
```

Comment:
Since there is an ```InventoryManager``` that handles all the Inventories in the World, the ```inventoryCreated``` event should be part of the ```InventoryManager```. Only the ```InventoryManager``` calls this method in world, and world only executes it when that method is call.